waiter/test/waiter/scaling_test.clj

```diff
@@ -667,45 +667,95 @@
                           "app4" (do
                                    (is (= -5 scale-amount))
                                    (is (= 10 scale-to-instances))
-                                   10)))
+                                   10)
+                          ; max instances is 20
+                          "app5" (do
+                                   (is (= 11 scale-amount))
+                                   (is (= 21 scale-to-instances))
+                                   18)))
         ; simple scaling function that targets outstanding-requests
-        test-scale-service (fn [{:strs [min-instances max-instances]} {:keys [total-instances outstanding-requests]}]
-                             (let [scale-to-instances (max min-instances (min max-instances outstanding-requests))]
+        test-scale-service (fn [{:strs [min-instances max-instances]}
+                                {:keys [expired-instances total-instances outstanding-requests]}]
+                             (let [target-instances (max min-instances (min max-instances outstanding-requests))
+                                   scale-to-instances (+ target-instances expired-instances)]
                                {:scale-to-instances scale-to-instances
-                                :target-instances scale-to-instances
+                                :target-instances target-instances
                                 :scale-amount (- scale-to-instances total-instances)}))]
-    (let [result (scale-services ["app1" "app2" "app3" "app4"]
+    (let [max-expired-unhealthy-instances-to-consider 2
+          result (scale-services ;; service-ids
+                                 ["app1" "app2" "app3" "app4" "app5"]
+                                 ;; service-id->service-description
                                  {"app1" (merge config {})
                                   "app2" (merge config {})
                                   "app3" (merge config {"min-instances" 5})
-                                  "app4" (merge config {"max-instances" 10})} ; service description
+                                  "app4" (merge config {"max-instances" 10})
+                                  "app5" (merge config {"max-instances" 20})
```

Comment:
Are we setting this to 20 to prove that we can scale beyond 20 to fix expired instances?  Comments describing the purpose/aspects of the test would help a lot.

Reply:
Added comment.